cassandra-management/src/main/java/org/hawkular/cassandra/management/CassandraMaintenanceJob.java

```diff
@@ -0,0 +1,130 @@
+import org.hawkular.rx.cassandra.driver.RxSession;
+import org.jboss.logging.Logger;
+
+import com.datastax.driver.core.Host;
+import com.datastax.driver.core.TableMetadata;
+import com.google.common.base.Stopwatch;
+import com.google.common.collect.ImmutableMap;
+
+import rx.Completable;
+import rx.Observable;
+import rx.functions.Func1;
+
+/**
+ * This job performs maintenance on the Cassandra cluster. Currently there is initial support for running cleanup, and
+ * we will add support for running anti-entropy repair. {@link CassandraManager} maintains an event log for relevant
+ * events in the Cassandra cluster, and the job looks at that log to determine what if any maintenance needs to be
+ * performed.
+ *
+ * @author jsanda
+ */
+public class CassandraMaintenanceJob implements Func1<JobDetails, Completable> {
+
+    private static Logger logger = Logger.getLogger(CassandraMaintenanceJob.class);
+
+    public static final String JOB_NAME = "CassandraMaintenanceJob";
+
+    public static final String CONFIG_ID = "org.hawkular.metrics.jobs." + JOB_NAME;
+
+    private ConfigurationService configurationService;
+
+    private EventDataAccess eventDataAccess;
+
+    private RxSession session;
+
+    public CassandraMaintenanceJob(RxSession session, EventDataAccess eventDataAccess,
+            ConfigurationService configurationService) {
+        this.session = session;
+        this.eventDataAccess = eventDataAccess;
+        this.configurationService = configurationService;
+    }
+
+    @Override
+    public Completable call(JobDetails jobDetails) {
+        logger.info("Starting Cassandra cluster maintenance");
+        Stopwatch stopwatch = Stopwatch.createStarted();
+
+        String jmxPort = jobDetails.getParameters().get("jmxPort");
+        String keyspace = jobDetails.getParameters().get("keyspace");
+
+        return eventDataAccess.findEventsForBucket(getCurrentBucket())
+                .collect(ClusterState::new, (state, event) -> {
+                    try {
+                        String hostname = event.getDetails().get("hostname");
+                        InetAddress address = InetAddress.getByName(hostname);
+                        state.setLastNodeAdded(address);
+                    } catch (UnknownHostException e) {
+                        throw new RuntimeException(e);
+                    }
+                })
+                .flatMap(state -> {
+                    if (state.getLastNodeAdded() == null) {
+                        return Observable.empty();
+                    }
+                    return getNodes().filter(address -> !address.equals(state.getLastNodeAdded()));
+                })
+                .concatMap(address -> createExecutionContexts(address, getTables(keyspace), jmxPort, keyspace))
+                .flatMap(context -> new CleanupCommand().run(context))
+                .toCompletable()
+                .concatWith(Completable.defer(() -> configurationService.save(CONFIG_ID, "lastRun",
```

Comment:
That actually won't help here. Our queries are basically hot observables since they are executed immediately without any subscription. The reason I use ```defer``` here is to ensure that ```configurationService.save``` is only called when the other Completable completes successfully.